src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/config/ClientTrackingConfigurationTest.kt

```diff
@@ -33,48 +35,51 @@ class ClientTrackingConfigurationTest {
 
   @BeforeEach
   fun setup() {
-    ThreadContext.setRequestTelemetryContext(RequestTelemetryContext(1L))
   }
 
   @AfterEach
   fun tearDown() {
-    ThreadContext.remove()
   }
 
   @Test
   fun shouldAddClientIdAndUserNameToInsightTelemetry() {
-
     // Given
     val token = jwtAuthHelper.createJwt("bob")
     val req = MockHttpServletRequest()
     req.addHeader(HttpHeaders.AUTHORIZATION, "Bearer $token")
     val res = MockHttpServletResponse()
 
+    val clientTrackingInterceptorSpy = spy(clientTrackingInterceptor)
```

Comment:
alternative approach is https://github.com/ministryofjustice/cmd-api/blob/021aac507ebf94e25bbc373bd271232cb50e3234/src/test/java/uk/gov/justice/digital/hmpps/cmd/api/config/ClientTrackingConfigurationTest.kt#L26

Reply:
will consider this for laters as this is a merge PR